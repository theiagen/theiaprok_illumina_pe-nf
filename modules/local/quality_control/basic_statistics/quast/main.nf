process QUAST {
    tag "$meta.id"
    label 'process_low'

    container "us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("${meta.id}_report.tsv"), emit: report
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def min_contig_length = params.quast_min_contig_length ?: 500
    """
    quast.py ${assembly} \\
        --output-dir . \\
        --threads ${task.cpus} \\
        --min-contig ${min_contig_length}
    
    mv report.tsv ${meta.id}_report.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        quast: \$(quast --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    touch ${meta.id}_report.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        quast: \$(quast --version)
    END_VERSIONS
    """
}
