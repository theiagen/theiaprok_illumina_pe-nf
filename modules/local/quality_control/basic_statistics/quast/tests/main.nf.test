nextflow_process {

    name "Test Process QUAST"
    script "../main.nf"
    process "QUAST"

    tag "modules"
    tag "quast"

    test("sarscov2 - noref") {

        when {
            process {
                """
                input[0] = [
                            [ id: 'test' ], 
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.report,
                    process.out.versions,
                    ).match() }
            )
        }
    }


    test("sarscov2 - all - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                            [ id: 'test1' ], 
                            file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true)
                ]          
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
