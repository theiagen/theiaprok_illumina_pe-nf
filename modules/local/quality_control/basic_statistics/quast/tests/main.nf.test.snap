{
    "sarscov2 - all - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_report.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": ["versions.yml:md5,b2801e0623da9725a7aebedb424aa8e3"]
                ,
                "report": [
                    [
                        {
                            "id": "test1"
                        },
                        "test1_report.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b2801e0623da9725a7aebedb424aa8e3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T17:34:37.428700068"
    },
    "sarscov2 - noref": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test_report.tsv:md5,06dd7886c52475c4f7c136f58b47348a"
                ]
            ],
            [
                "versions.yml:md5,b2801e0623da9725a7aebedb424aa8e3"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T17:26:40.173244079"
    }
}